src/NexusMods.App.UI/Pages/CollectionDownload/CollectionDownloadViewModel.cs

```diff
@@ -367,17 +367,28 @@ public CollectionDownloadViewModel(
             R3.Observable.Return(collectionJsonFile)
                 .ObserveOnThreadPool()
                 .SelectAwait((jsonFile, cancellationToken) => nexusModsLibrary.ParseCollectionJsonFile(jsonFile, cancellationToken))
-                .Select(static collectionRoot => collectionRoot.Info.InstallInstructions)
-                .Where(static instructions => !string.IsNullOrWhiteSpace(instructions))
                 .ObserveOnUIThreadDispatcher()
-                .Subscribe((this, serviceProvider), static (instructionsText, state) =>
+                .Subscribe((this, serviceProvider), static (collectionRoot, state) =>
                 {
                     var (self, serviceProvider) = state;
 
+                    var collectionInstructionsText = collectionRoot.Info.InstallInstructions;
+
+                    var modsInstructions = collectionRoot.Mods
+                        .Select(static mod => (mod.Name, mod.Instructions, mod.Optional))
+                        .Where(static tuple => !string.IsNullOrWhiteSpace(tuple.Instructions))
+                        .Select(static tuple => new ModInstructions(tuple.Name, tuple.Instructions, tuple.Optional ? CollectionDownloader.ItemType.Optional : CollectionDownloader.ItemType.Required))
+                        .ToArray();
+
+                    var optionalModsInstructions = modsInstructions.Where(static x => x.ItemType == CollectionDownloader.ItemType.Optional).ToArray();
+                    var requiredModsInstructions = modsInstructions.Where(static x => x.ItemType == CollectionDownloader.ItemType.Required).ToArray();
+
                     var markdownRendererViewModel = serviceProvider.GetRequiredService<IMarkdownRendererViewModel>();
-                    markdownRendererViewModel.Contents = instructionsText;
+                    markdownRendererViewModel.Contents = collectionInstructionsText;
 
                     self.InstructionsRenderer = markdownRendererViewModel;
+                    self.RequiredModsInstructions = requiredModsInstructions;
+                    self.OptionalModsInstructions = optionalModsInstructions;
```

Comment:
The markdown renderer is always created, even if there is nothing to display which means that the codebehind will then show the empty tab even if there is nothing.
Add a check here before creating and setting the renderer.